frontends/concrete-python/concrete/fhe/mlir/processors/assign_bit_widths.py

```diff
@@ -56,52 +56,53 @@ def __init__(
         self.multivariate_strategy_preference = multivariate_strategy_preference
         self.min_max_strategy_preference = min_max_strategy_preference
 
-    def apply(self, graph: Graph):
+    def apply_many(self, graphs: Dict[str, Graph]):
         optimizer = z3.Optimize()
 
         max_bit_width: z3.Int = z3.Int("max")
         bit_widths: Dict[Node, z3.Int] = {}
 
-        additional_constraints = AdditionalConstraints(
-            optimizer,
-            graph,
-            bit_widths,
-            self.comparison_strategy_preference,
-            self.bitwise_strategy_preference,
-            self.shifts_with_promotion,
-            self.multivariate_strategy_preference,
-            self.min_max_strategy_preference,
-        )
+        for _, graph in graphs.items():
+            additional_constraints = AdditionalConstraints(
+                optimizer,
+                graph,
+                bit_widths,
+                self.comparison_strategy_preference,
+                self.bitwise_strategy_preference,
+                self.shifts_with_promotion,
+                self.multivariate_strategy_preference,
+                self.min_max_strategy_preference,
+            )
 
-        nodes = graph.query_nodes(ordered=True)
-        for i, node in enumerate(nodes):
-            assert isinstance(node.output.dtype, Integer)
-            required_bit_width = node.output.dtype.bit_width
+            nodes = graph.query_nodes(ordered=True)
+            for i, node in enumerate(nodes):
+                assert isinstance(node.output.dtype, Integer)
+                required_bit_width = node.output.dtype.bit_width
 
-            bit_width_hint = node.properties.get("bit_width_hint")
-            if bit_width_hint is not None:
-                required_bit_width = max(required_bit_width, bit_width_hint)
+                bit_width_hint = node.properties.get("bit_width_hint")
+                if bit_width_hint is not None:
+                    required_bit_width = max(required_bit_width, bit_width_hint)
 
-            bit_width = z3.Int(f"%{i}")
-            bit_widths[node] = bit_width
+                bit_width = z3.Int(f"%{i}")
```

Comment:
Lastly this would mean changing how the constraints are printed in `graph.py`, maybe we can separate them using a blank space between function and composition constraints.

Reply:
We could name composition constraints like:
```
+ add.%1 -> square.%0
```
Wires would be easy to debug in the future as well!

Reply:
Will do